### PR TITLE
machined: unprivileged machine registration

### PR DESCRIPTION
**Pull request by poettering**
This adds the ability for unpriv vmspawn/nspawn to register with the system-wide machined, and thus get benefit from name resolution and similar things.

Contains numerous others fixes and additions.

### CODE REVIEW

**Comment by YHNdnzj** [comment]
Add "Replaces https://github.com/systemd/systemd/pull/15312" to description/commit msg?

File: src/machine/machined-dbus.c [hunk 1]
```diff
@@ -283,21 +283,19 @@ static int method_create_or_register_machine(
         if (leader == 0) {
                 _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
 
-                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID, &creds);
+                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID|SD_BUS_CREDS_PIDFD, &creds);
                 if (r < 0)
-                        return r;
+                        return sd_bus_error_set_errnof(error, r, "Failed to acquire clien credentials: %m");
```
Comment (daandemeyer) [inline src/machine/machined-dbus.c]: "clien" => "client"

**Comment by poettering** [comment]
> Add "Replaces #15312" to description/commit msg?

I think we aren't ready for that yet, that PR did a lot more. I need some time to dissect that, and see what else need to look into.

**Review by bluca — APPROVED** [review]

File: src/vmspawn/vmspawn.c [hunk 2]
```diff
@@ -2229,8 +2239,10 @@ static int verify_arguments(void) {
         if (!strv_isempty(arg_initrds) && !arg_linux)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Option --initrd= cannot be used without --linux=.");
 
-        if (arg_register && !arg_privileged)
-                return log_error_errno(SYNTHETIC_ERRNO(EPERM), "--register= requires root privileges, refusing.");
+        if (arg_keep_unit && arg_register && cg_pid_get_owner_uid(0, NULL) >= 0)
```
Comment (YHNdnzj) [inline src/vmspawn/vmspawn.c:2242]: I don't think the check is enough? If vmspawn is running as a user service, which in turn has a user slice unit, we would accept it and `machine_stop` would eventually kill `user@.service`. We need to check this properly in server side and tighten the check on client side.

**Pull request merged**